shared-bindings/bitmaptools/__init__.c

```diff
@@ -868,7 +868,92 @@ STATIC mp_obj_t bitmaptools_dither(size_t n_args, const mp_obj_t *pos_args, mp_m
     return mp_const_none;
 }
 MP_DEFINE_CONST_FUN_OBJ_KW(bitmaptools_dither_obj, 0, bitmaptools_dither);
+// requires all 5 arguments
 
+//| def draw_circle(
+//|     dest_bitmap: displayio.Bitmap, x: int, y: int, radius: int, value: int
+//| ) -> None:
+//|     """Draws a circle into a bitmap specified using a center (x0,y0) and radius r.
+//|
+//|     :param bitmap dest_bitmap: Destination bitmap that will be written into
+//|     :param int x: x-pixel position of the circle's center
+//|     :param int y: y-pixel position of the circle's center
+//|     :param int radius: circle's radius
+//|     :param int value: Bitmap palette index that will be written into the
+//|            circle in the destination bitmap
+//|
+//|     .. code-block:: Python
+//|
+//|        import board
+//|        import displayio
+//|        import bitmaptools
+//|
+//|        display = board.DISPLAY
+//|        main_group = displayio.Group()
+//|        display.root_group = main_group
+//|
+//|        palette = displayio.Palette(2)
+//|        palette[0] = 0xffffff
+//|        palette[1] = 0x440044
+//|
+//|        bmp = displayio.Bitmap(128,128, 2)
+//|        bmp.fill(0)
+//|
+//|        bitmaptools.circle(64,64, 32, 1)
+//|
+//|        tilegrid = displayio.TileGrid(bitmap=bmp, pixel_shader=palette)
+//|        main_group.append(tilegrid)
+//|
+//|        while True:
+//|            pass
+//|
+//|     """
+//|
+//|     ...
+//|
+STATIC mp_obj_t bitmaptools_obj_draw_circle(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum {ARG_dest_bitmap, ARG_x, ARG_y, ARG_radius, ARG_value};
+
+    static const mp_arg_t allowed_args[] = {
+        {MP_QSTR_dest_bitmap, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL}},
+        {MP_QSTR_x0, MP_ARG_REQUIRED | MP_ARG_INT, {.u_obj = MP_OBJ_NULL}},
+        {MP_QSTR_y0, MP_ARG_REQUIRED | MP_ARG_INT, {.u_obj = MP_OBJ_NULL}},
+        {MP_QSTR_radius, MP_ARG_REQUIRED | MP_ARG_INT, {.u_obj = MP_OBJ_NULL}},
+        {MP_QSTR_value, MP_ARG_REQUIRED | MP_ARG_INT, {.u_obj = MP_OBJ_NULL}},
```

Comment:
1. Finish changing `x0` and `y0` to `x` and `y`.
2. `MP_ARG_REQUIRED` means you can omit the default value, since it will never be used.
```suggestion
        {MP_QSTR_x, MP_ARG_REQUIRED | MP_ARG_INT},
        {MP_QSTR_y, MP_ARG_REQUIRED | MP_ARG_INT},
        {MP_QSTR_radius, MP_ARG_REQUIRED | MP_ARG_INT},
        {MP_QSTR_value, MP_ARG_REQUIRED | MP_ARG_INT},
```